### PR TITLE
Make grouping query more efficient, fix bulk actions bug

### PR DESCRIPTION
**Pull request by kewitham**
[Issue #1527](https://github.com/usdoj-crt/crt-portal-management/issues/1527)
[Issue #1524](https://github.com/usdoj-crt/crt-portal-management/issues/1524)
[Issue #1528](https://github.com/usdoj-crt/crt-portal-management/issues/1528)

## What does this change?
This updates the grouping query to make it more efficient in the following ways:
1. Removing the count() query for each group
2. Applying the filters before grouping to reduce the number of reports to group
3. Removing an extra unnecessary grouping query

It also fixes the bulk actions in groups so they only apply to members of the group.

### Author

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [ ] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/views.py [hunk 1]
```diff
@@ -764,9 +765,13 @@ def post(self, request):
         selected_all = request.POST.get('all', '') == 'all'
         confirm_all = request.POST.get('confirm_all', '') == 'confirm_all'
         ids = request.POST.get('ids', '').split(',')
+        query_string = request.POST.get('query_string', None)
+
+        if query_string is None:
+            query_string = return_url_args
```
Comment (okkays) [inline crt_portal/cts_forms/views.py:771]:
```suggestion
        query_string = request.POST.get('query_string', return_url_args)
```

File: crt_portal/cts_forms/views.py [hunk 2]
```diff
@@ -244,7 +244,7 @@ def render_group_view(request, profile_form, selected_assignee_id, selected_camp
             })
     else:
         updated_group_params = group_params
-    final_data = get_group_view_data(request, updated_group_queries[0]["qs"], filters, grouping, updated_group_params[0], 'All other reports')
+    final_data = group_view_data[len(group_view_data) - 1]["data"]
```
Comment (okkays) [inline crt_portal/cts_forms/views.py:247]:
```suggestion
    final_data = group_view_data[-1]["data"]
```

**Review by okkays — CHANGES_REQUESTED** [review]
A few quick changes to make, but mostly lgtm

Can you explain in words how this makes things more efficient, and what kinds of improvements you're seeing on dev? (how do you know that this will work)

**Review by okkays — APPROVED** [review]
LGTM!

**Review by ilodidoj — APPROVED** [review]
@kewitham ready for dev merge.

**Pull request merged**